lib/textinterceptor/src/main/java/eu/kanade/tachiyomi/lib/textinterceptor/TextInterceptor.kt

```diff
@@ -18,69 +18,73 @@ import okhttp3.Response
 import okhttp3.ResponseBody.Companion.toResponseBody
 import java.io.ByteArrayOutputStream
 
+// Designer values:
+private const val WIDTH: Int = 1000
+private const val X_PADDING: Float = 50f
+private const val Y_PADDING: Float = 25f
+private const val HEADING_FONT_SIZE: Float = 36f
+private const val BODY_FONT_SIZE: Float = 30f
+private const val SPACING_MULT: Float = 1.1f
+private const val SPACING_ADD: Float = 2f
+
+// No need to touch this one:
+private const val HOST = TextInterceptorHelper.HOST
+
 class TextInterceptor : Interceptor {
     // With help from:
     // https://github.com/tachiyomiorg/tachiyomi-extensions/pull/13304#issuecomment-1234532897
     // https://medium.com/over-engineering/drawing-multiline-text-to-canvas-on-android-9b98f0bfa16a
-
-    companion object {
-        // Designer values:
-        private const val WIDTH: Int = 1000
-        private const val X_PADDING: Float = 50f
-        private const val Y_PADDING: Float = 25f
-        private const val HEADING_FONT_SIZE: Float = 36f
-        private const val BODY_FONT_SIZE: Float = 30f
-        private const val SPACING_MULT: Float = 1.1f
-        private const val SPACING_ADD: Float = 2f
-
-        // No need to touch this one:
-        private const val HOST = TextInterceptorHelper.HOST
-    }
-
     override fun intercept(chain: Interceptor.Chain): Response {
         val request = chain.request()
         val url = request.url
         if (url.host != HOST) return chain.proceed(request)
 
-        val creator = textFixer("Author's Notes from ${url.pathSegments[0]}")
-        val story = textFixer(url.pathSegments[1])
+        val heading = url.pathSegments[0].takeIf { it.isNotEmpty() }?.let {
+            val creator = textFixer("Author's Notes from ${url.pathSegments[0]}")
```

Comment:
nitpick: "Author's Note" text should come from the source, so this lib can be used as general purpose text writing lib